torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -20,10 +20,32 @@
     _ATEN_OP_OR_TORCH_FN_TABLE,
     _register_layout_cls,
     _get_layout_tensor_constructor,
+    LayoutType,
 )
+from typing import ClassVar
+from dataclasses import dataclass
 
 aten = torch.ops.aten
 
+@dataclass(frozen=True)
```

Comment:
not sure what benefit we're getting from using dataclasses, what you want seems to be closer to an enum with 2 values Plain or TensorCoreTiled and specifically if using TensorCoreTiled there's a util function called pad_input that doesn't have to be part of the class